src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -667,40 +666,65 @@ export function getInputType(
             logger.info(
                 `No specified type for discriminator property '${model.DiscriminatorPropertyName}'. Assume it is a string.`
             );
-            const discriminatorProperty = {
+            const discriminatorProperty: InputModelProperty = {
                 Name: model.DiscriminatorPropertyName,
                 SerializedName: model.DiscriminatorPropertyName,
                 Description: "Discriminator",
                 IsRequired: true,
                 IsReadOnly: false,
-                IsNullable: false,
```

Comment:
this is removed because we never have a `IsNullable` property in `InputModelProperty` type.